sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/LockRenewalOperation.java

```diff
@@ -0,0 +1,212 @@
+        this.renewalOperation = Objects.requireNonNull(renewalOperation, "'renewalOperation' cannot be null.");
+        this.isSession = isSession;
+
+        Objects.requireNonNull(lockedUntil, "'lockedUntil cannot be null.'");
+        Objects.requireNonNull(maxLockRenewalDuration, "'maxLockRenewalDuration' cannot be null.");
+
+        if (maxLockRenewalDuration.isNegative()) {
+            throw logger.logThrowableAsError(new IllegalArgumentException(
+                "'maxLockRenewalDuration' cannot be negative."));
+        }
+
+        this.lockedUntil.set(lockedUntil);
+        this.subscription = getRenewLockOperation(lockedUntil, maxLockRenewalDuration);
+    }
+
+    /**
+     * Gets the current instant the message or session is locked until.
+     *
+     * @return the instant the message or session is locked until.
+     */
+    public Instant getLockedUntil() {
+        return lockedUntil.get();
+    }
+
+    /**
+     * Gets the message lock token for the renewal operation.
+     *
+     * @return The message lock token or {@code null} if a session is being renewed instead.
+     */
+    public String getLockToken() {
+        return isSession ? null : lockToken;
+    }
+
+    /**
+     * Gets the session id for this lock renewal operation.
+     *
+     * @return The session id or {@code null} if it is not a session renewal.
+     */
+    public String getSessionId() {
+        return isSession ? lockToken : null;
+    }
+
+    /**
+     * Gets the current status of the renewal operation.
+     *
+     * @return The current status of the renewal operation.
+     */
+    public LockRenewalStatus getStatus() {
+        return status.get();
+    }
+
+    /**
+     * Gets the exception if an error occurred whilst renewing the message or session lock.
+     *
+     * @return the exception if an error occurred whilst renewing the message or session lock, otherwise {@code null}.
+     */
+    public Throwable getThrowable() {
+        return throwable.get();
+    }
+
+    /**
+     * Cancels the lock renewal operation.
+     */
+    @Override
+    public void close() {
+        if (isDisposed.getAndSet(true)) {
+            return;
+        }
+
+        if (status.compareAndSet(LockRenewalStatus.RUNNING, LockRenewalStatus.CANCELLED)) {
+            logger.verbose("token[{}] Cancelled operation.", lockToken);
+        }
+
+        cancellationProcessor.onComplete();
+        subscription.dispose();
+    }
+
+    /**
+     * Gets the lock renewal operation. if the {@code maxLockRenewalDuration} is {@link Duration#isZero()}, then the
+     * lock is never renewed.
+     *
+     * @param initialLockedUntil When the initial call is locked until.
+     * @param maxLockRenewalDuration Duration to renew lock for.
+     * @return The subscription for the operation.
+     */
+    private Disposable getRenewLockOperation(Instant initialLockedUntil, Duration maxLockRenewalDuration) {
+        if (maxLockRenewalDuration.isZero()) {
```

Comment:
isZero() | isNegative() ?

Reply:
IsNegative is an invalid operation. I'll throw before it even gets here.